### PR TITLE
Password login controller

### PR DESCRIPTION
**Pull request by jacksonargo**
Password login controller provides an api for establishing login sessions via form values `user` and `pass`.

This also adds a view so users can login from web browser to establish the session.

https://jackson.vvgo.org/login

### CODE REVIEW

File: pkg/api/login.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		unauthorized(w)
+		return
+	}
+
+	identity = login.Identity{
+		Kind:  login.KindPassword,
+		Roles: gotRoles,
+	}
+
+	cookie, err := x.Sessions.NewCookie(ctx, &identity, 31557600*time.Second)
```
Comment (0az) [inline pkg/api/login.go:52]: Where does `31557600` come from?

Comment (jacksonargo) [reply pkg/api/login.go:52]: Oh ya I think that's a years worth of seconds haha. This should probably be an env var or something.

**Comment by jacksonargo** [comment]
Passwords are fed into the controller at startup via environment variables. So, no database lookups need to happen right now.

What are the concerns around hashing? What needs to be salted+hashed and where?

**Comment by jacksonargo** [comment]
I want to avoid making user accounts for individual users. The scope of this login controller is basically to replace the basic-auth with an actual http view. This will also be useful for people who cant login via oauth.

File: public/css/login.css [hunk 2]
```diff
@@ -0,0 +1,34 @@
+.form-signin {
```
Comment (0az) [inline public/css/login.css:1]: I'd personally write this as `#form-login`.

**Pull request merged**